microsoft/testsuites/firewalld/firewalldsuite.py

```diff
@@ -0,0 +1,229 @@
+    )
+
+
+@TestSuiteMetadata(
+    area="firewalld",
+    category="functional",
+    description="""
+    This test suite validates firewalld.
+    It is a collection of tests which run against
+    a local firewalld installation.
+    It runs isolated inside temporary network namespaces.
+    These tests interact with both iptables & nftables as backend.
+    The testsuite is provided by the firewalld-test rpm.
+    """,
+)
+class FirewalldSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner) or node.os.information.version < "3.0.0":
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "Firewalld testsuite is supported only on AzureLinux 3.0."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""
+        This test case runs the complete testsuite.
+        The set of tests include
+        1. cli: firewall-cmd client tests
+        2. dbus: dbus interface tests
+        3. python: python binding tests
+        4. features: firewalld daemon functional tests
+        5. regressions: regression tests
+        """,
+        priority=3,
+    )
+    def verify_firewalld(
+        self,
+        node: Node,
+        log: Logger,
+        result: TestResult,
+    ) -> None:
+        # Install the dependencies for running the test suite
+        if isinstance(node.os, CBLMariner):
+            node.os.install_packages(
+                [
+                    "ebtables",
+                    "nftables",
+                    "python3-dbus",
+                    "iproute",
+                    "firewalld",
+                    "firewalld-test",
+                ]
+            )
+
+        # Check if ipv6_rpfilter config is supported
+        # Note: Right now, this is the only known config which
+        # causes failure in starting firewalld daemon.
+        if not _is_ipv6_rpfilter_supported(node):
+            raise SkippedException(
+                "Skipping tests. Needs kernel config CONFIG_FIB_INET &"
+                "CONFIG_FIB_IPV6 enabled to use fib based expressions for "
+                "ipv6_rpfilter configuration."
+            )
+
+        # these paths are specific to testsuite.
+        test_suite_dir = "/usr/share/firewalld/testsuite"
+        test_suite_binary = "/usr/share/firewalld/testsuite/testsuite"
+        test_logs_dir = "/usr/share/firewalld/testsuite/testsuite.dir"
+        test_log = "/usr/share/firewalld/testsuite/testsuite.log"
+
+        # these commands are specific to testsuite binary.
+        # arg to set log directory
+        set_logs_dir_arg = f"-C {test_suite_dir}"
+
+        # arg to clean test artifacts
+        clean_arg = "-c"
+
+        # arg to set number of jobs
+        jobs_arg = "-j4"
+
+        clean_command = f"{test_suite_binary} {clean_arg} {set_logs_dir_arg}"
```

Comment:
Move this line just before `node.execute(clean_command, sudo=True, shell=False)` to make it clearer. Do the same for `test_command`.